jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -414,6 +414,40 @@ public String toString()
             " eof=" + _consumedEof;
     }
 
+    /**
+     * <p>{@link Content} interceptor that can be registered using {@link #setInterceptor(Interceptor)} or
+     * {@link #addInterceptor(Interceptor)}.
+     * When {@link Content} instances are generated, they are passed to the registered interceptor (if any)
+     * that is then responsible for providing the actual content that is consumed by {@link #read(byte[], int, int)} and its
+     * sibling methods.</p>
+     * <p>A minimal implementation could be as simple as:
+     * <pre>
+     * public HttpInput.Content readFrom(HttpInput.Content content)
+     * {
+     *     LOGGER.debug("read content: {}", asString(content));
+     *     return content;
+     * }
+     * </pre>
+     * which would not do anything with the content besides logging it. A more involved implementation could look like the
+     * following:
+     * <pre>
+     * public HttpInput.Content readFrom(HttpInput.Content content)
+     * {
+     *     if (content.hasContent())
+     *         processContent(content.getByteBuffer());
+     *     if (content.isEof())
+     *         endProcessing();
+     *     return processedContent();
```

Comment:
This example is missing is `isSpecial()` call, which is the one actually used in the implementations.

I would suggest:

```suggestion
     *     if (content.hasContent())
     *         this.processedContent = processContent(content.getByteBuffer());
     *     if (content.isEof())
     *         disposeResources();
     *     return content.isSpecial() ? content : this.processedContent;
```

Reply:
Okay, updated javadoc.

Reply:
Wait, does this mean `content.isSpecial()` is mutually exclusive to `content.hasContent()` always?

You'll never have a scenario where `(content.hasContent() && content.isSpecial())` is true?

Reply:
@joakime at least, the JavaDoc on `HttpInput.hasContent()` says so:

> Always false on a special content.

Reply:
@joakime a `Content` is special when it doesn't hold bytes but something more akin to a special event: EOF or an error. So yes, as was pointed out, the `Content` contract explicitly states that `hasContent()` returns `false` when content is special. Other methods also document specific behavior when the content is special.

I've modified the `Content.isSpecial()` javadoc to better explain this.